### ISSUE TITLE
Erroneous 'Cannot access member for type "NoReturn"' in pandas

### ISSUE THREAD

**Issue by betaprior** [microsoft/pyright#1974]
The following snippet results in a type-checking error, even though no actual type errors are present.
The type error is apparently due to the usage of `pd.DataFrame.drop` method.

If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
import pandas as pd
df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
df = df.drop(columns=['A'])  # removing this makes type error go away
df0 = df[df.B == 4]
print(df.B)
print(df0.B)  # error: Cannot access member "B" for type "NoReturn"
```

**VS Code extension or command-line**
Observing this behavior using emacs lsp-mode, but easiest to reproduce on command line:
`pyright --lib myfile.py`

**Additional context**

Pyright 1.1.148 with pandas 1.2.4 and Python 3.8.5

**Comment by erictraut** [comment]
Where are you getting your types for pandas? Are you using type stubs?

It looks like you're using the `--lib` option, which attempts to infer types from the library code. This option infers types that are often good enough for interactive completion suggestions, but they're typically not good enough for type checking. If you want to use type checking, you should find type stubs for the libraries you are using.

**Comment by betaprior** [comment]
Thanks for the explanation. Using stubs from `python-type-stubs` repo and following instructions in https://github.com/microsoft/python-type-stubs/issues/44 solved the issue. I have to say, there seems to be quite a lack of good information about best practices for using type stubs (for example, there are at least 3 repos that provide stubs for pandas, including `data-science-types` and `pandas-stubs`). In addition, the defaults that ship with emacs lsp-pyright mode (i.e. using `--lib` option result in these false positives, which is very confusing. Perhaps it would be helpful to warn about this behavior in the appropriate section of the README.
